src/main/java/atlas/parser/Parser.java

```diff
@@ -0,0 +1,138 @@
+package atlas.parser;
+
+import java.time.DateTimeException;
+
+import atlas.commands.AllCommand;
+import atlas.commands.Command;
+import atlas.commands.DeadlineCommand;
+import atlas.commands.DeleteCommand;
+import atlas.commands.EventCommand;
+import atlas.commands.ExitCommand;
+import atlas.commands.FindCommand;
+import atlas.commands.ListCommand;
+import atlas.commands.MarkCommand;
+import atlas.commands.TodoCommand;
+import atlas.commands.UnmarkCommand;
+import atlas.exceptions.AtlasException;
+import atlas.utils.DateTime;
+
+/**
+ * Parser class which contain methods to parse the command typed into the chatbot by the user.
+ */
+public class Parser {
+    /**
+     * Parses the command typed into the chatbot by the user.
+     *
+     * @param fullCommand The full command a user types into the chatbot through the command line.
+     * @return Command The command to be executed by the chatbot.
+     * @throws AtlasException The exception to be thrown in the event of any error.
+     */
+    public static Command parse(String fullCommand) throws AtlasException {
+        String[] commandsArray = fullCommand.split(" ");
+        String command = commandsArray[0].toUpperCase();
+        try {
+            switch (AllCommand.valueOf(command)) {
+            case BYE:
+                return new ExitCommand();
+            case LIST:
+                if (commandsArray.length > 1) {
+                    throw new AtlasException("List command should not be called with any additional arguments.");
+                }
+
+                return new ListCommand();
+            case MARK:
+                if (commandsArray.length == 1) {
+                    throw new AtlasException("Marking a task as done requires the task number.");
+                } else if (isNotNumber(commandsArray[1])) {
+                    throw new AtlasException("Task number provided is not a number.");
+                } else if (commandsArray.length > 2) {
+                    throw new AtlasException("Marking a task as done only requires the task number.");
```

Comment:
Perhaps throwing or handling these exceptions would be more suitable if done inside the MarkCommand class?